modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainer.java

```diff
@@ -56,10 +55,14 @@ public MySQLContainer(final DockerImageName dockerImageName) {
         addExposedPort(MYSQL_PORT);
     }
 
+    /**
+     * @deprecated use {@link #getLivenessCheckPortNumbers()} instead
+     */
     @NotNull
     @Override
+    @Deprecated
     protected Set<Integer> getLivenessCheckPorts() {
-        return new HashSet<>(getMappedPort(MYSQL_PORT));
+        return super.getLivenessCheckPorts();
```

Comment:
Previous changes was binary incompatible - so can't remove this method. Also marked this method as deprecated to discourage usage of this method.

Reply:
Ah yes, you are right. Did you already have a check through the whole code base to see if there are other non-sensical overrides? Indeed, I can't imagine a good use case for an override (they might be here for legacy reasons).

Reply:
Ok I've gone through and replaced all non-sensical overrides. Some of which were also not implemented correctly as above. Also marked those overrides as deprecated.